.github/workflows/qe-tests.yml

```diff
@@ -0,0 +1,114 @@
+name: QE Tests
+
+on: [push, pull_request]
+
+jobs:
+    # "Build" job
+    build:
+      # This job tries to build PKI from src on a fresh docker container.
+      # The docker container is spawned by github itself and we merely just
+      # run the build commands. We then upload the artifact for consumption
+      # by the test jobs + for the public to consume. This job **does not**
+      # upload any build logs as they are visible in the log console itself.
+
+      name: Build PKI
+      runs-on: ubuntu-latest
+      container: fedora:${{ matrix.os }}
+      strategy:
+          matrix:
+            os: ['31', '32']
+      steps:
+          - name: Update base image
+            run: |
+                  set -x &&
+                  dnf update -y &&
+                  dnf install -y dnf-plugins-core rpm-build git
+
+          - name: Clone the repository
+            uses: actions/checkout@v2
+
+          - name: Install PKI build deps
+            run: |
+                  dnf copr enable -y @pki/master
+                  dnf builddep -y --allowerasing --spec ./pki.spec
+
+          - name: Build PKI packages
+            run: ./build.sh --with-timestamp --with-commit-id --work-dir=../packages rpm
+
+          - name: Compress RPMS
+            run: tar -czf pki-rpms.tar.gz ../packages/RPMS/*
+
+          # upload-artifact runs on host-vm rather than inside the container. Fixed in v2 (unreleased)
+          # Bug: https://github.com/actions/upload-artifact/issues/13#issuecomment-532936650
+          - name: Upload RPM artifacts
+            uses: actions/upload-artifact@v1
+            with:
+              name: pki-build-${{ matrix.os }}
+              path: pki-rpms.tar.gz
+
+    # Tier 0
+    installation-sanity:
+      # This job uses Ansible playbooks in the tests/ dir to setup a PKI deployment.
+      # All 5 subsystems are deployed on "discrete" instances
+      name: installation-sanity
+      needs: build
+      runs-on: macos-latest
+      env:
+        CONTAINER: pkitest
+        BUILDDIR: /tmp/workdir
+        PKIDIR: /tmp/workdir/pki
+        LOGS: ${GITHUB_WORKSPACE}/logs.txt
+        COPR_REPO: "@pki/master"
+        HOSTFILE: /tmp/hostfile
+        CONTROLLER_IP: "192.168.33.10"
+        MASTER_IP: "192.168.33.20"
```

Comment:
Could we remove the hard-coded IP addresses from the source?

Reply:
No,we can't for the following reasons:

1. Our playbooks need an IP address (127.0.0.1 does not work) to execute the deployment scripts
2. Vagrant by default does not assign a address. Instead, it uses 127.0.0.1 with a custom ssh port to access the VMs. So, we need to specify a valid hard-coded private network ip in Vagrantfile. 
3. The inventory file supplied to play books can be generated automatically. But, the autogenerated inventory is not reliable (see note 2 in PR description). So, we use a static inventory with a hard-coded file. 

Since these are local VMs and not reliant on any external labs/infra, these hard codes IPs are harmless :-)

My $0.02 ;-)

Reply:
OK, as long as it's not pointing to a real machine in internal network.

Reply:
I don't think there are any issues so far as I chose a random local IP. Even if there is a collision, this variable is declared as a global variable so updating it will fix in all places :-)